phalcon/db/adapter/pdo.zep

```diff
@@ -85,9 +85,12 @@ abstract class Pdo extends Adapter
 	 */
 	public function connect(descriptor = null)
 	{
-		var username, password, dsnParts, dsnAttributes,
+		var eventsManager,
+			username, password, dsnParts, dsnAttributes,
 			persistent, options, key, value;
 
+		let eventsManager = <ManagerInterface> this->_eventsManager;
```

Comment:
the `\Phalcon\Db\Adapter\Pdo` class extends from `\Phalcon\Db\Adapter`, but `\Phalcon\Db\Adapter` implements `\Phalcon\Events\EventsAwareInterface`. So you can modify `\Phalcon\Events\EventsAwareInterface:: getEventsManager` as follows:

```zephir
public function getEventsManager() -> <ManagerInterface>
{
    var eventsManager;

    let eventsManager = this->_eventsManager;
    if typeof eventsManager != "object" {
        let eventsManager = new \Phalcon\Events\Manager();

        // Or 
        // var di = \Phalcon\Di::getDefault();
        // let eventsManager = di->getShared('eventsManager');

        let this->_eventsManager = eventsManager;
    }

    return eventsManager;
}
```